src/main/battlecode/world/RobotControllerImpl.java

```diff
@@ -812,15 +812,15 @@ public boolean canChop(MapLocation loc) {
         boolean correctType = (getType() == RobotType.LUMBERJACK);
         boolean canInteract = canInteractWithTree(loc);
 
-        return correctType && canInteract;
+        return correctType && canInteract && !hasAttacked();
```

Comment:
Chop counts as an attack though, so I believe this is proper form. shake() and water() have separate counts, so they have those parameter-less methods.